operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/SSABasedGenericKubernetesResourceMatcher.java

```diff
@@ -104,7 +104,9 @@ public boolean matches(R actual, R desired, Context<?> context) {
     removeIrrelevantValues(desiredMap);
 
     if (LoggingUtils.isNotSensitiveResource(desired)) {
-      log.debug("Pruned actual: \n {} \n desired: \n {} ", prunedActual, desiredMap);
+      var actualYaml = objectMapper.asYaml(prunedActual);
```

Comment:
I would protect these calls with a check to whether debug is enabled so as to not inferred the YAML serialization cost in that situation.

Reply:
> I would protect these calls with a check to whether debug is enabled so as to not inferred the YAML serialization cost in that situation.

Reply:
@metacosm 
https://github.com/operator-framework/java-operator-sdk/pull/2542/commits/72911d651efe4268dade07d05e5a5377d1c28ab3

fixed it in this commit

thank you!!